### PR TITLE
Release build scripts for Metal support

### PR DESCRIPTION
**Pull request by petr-pokorny-1**
This PR integrates MetalANGLE framework into release build scripts and fixes iOS test app and Benchmark app.

### CODE REVIEW

**Review by sulewicz — APPROVED** [review]
Looks great!

**Comment by roblabs** [comment]
Approved and here are my testing results.  No changes required to perform archiving and distro on TestFlight, good job @petr-pokorny-1 and I say its **ready for you to merge into `maplibre:metal-support`.**

1. - [x] The Swift Package Manager distribution repo now has a **Swift Playground** that allows developers to quickly test a map.  See that repo & PR: https://github.com/maplibre/maplibre-gl-native-distribution/pull/5

---

2. - [x] Archive & distribute in a private **TestFlight**
    - [ ] Repeat Archiving & TestFlight with and without `MetalANGLE`

*Archiving test of the `iosapp` with MapLibre + MetalANGLE*
DEVICE TYPE	 | DOWNLOAD SIZE - This is the compressed size of the app downloaded over the air. | INSTALL SIZE - This is the amount of disk space the app will take up on the customer's device.
:-- | :--: | :--:
iPhone 12 Pro Max | 5047 KB | 20712 KB
iPhone 7 Plus | 12385 KB | 20915 KB

*Estimated App Store file sizes and descriptions from TestFlight at App Store Connect*

File: platform/ios/platform/ios/scripts/deploy-swift-package.sh [hunk 1]
```diff
@@ -106,11 +106,17 @@ make xcframework
 step "Zipping xcframeworks…"
 MAPBOX_ZIP_FILE="Mapbox-${PUBLISH_VERSION}.zip"
 (cd ${BINARY_DIRECTORY} && rm -f ${MAPBOX_ZIP_FILE} && zip -yr ${MAPBOX_ZIP_FILE} Mapbox.xcframework)
+METAL_ANGLE_ZIP_FILE="MetalANGLE-${PUBLISH_VERSION}.zip"
+(cd ${BINARY_DIRECTORY} && rm -f ${METAL_ANGLE_ZIP_FILE} && zip -yr ${METAL_ANGLE_ZIP_FILE} MetalANGLE.xcframework)
```
Comment (roblabs) [inline platform/ios/platform/ios/scripts/deploy-swift-package.sh:110]: The version numbers of the binaries are as below, but I think this script pulls the `{PUBLISH_VERSION}` from the git tags.  If we cannot tag the MetalANGLE, then we may be able to use the binaries version as it is on disk.

image

File: platform/ios/platform/ios/scripts/swift_package_template.swift [hunk 2]
```diff
@@ -6,14 +6,21 @@ let package = Package(
...
     targets: [
         .binaryTarget(
             name: "Mapbox",
             url: "MAPBOX_PACKAGE_URL",
-            checksum: "MAPBOX_PACKAGE_CHECKSUM")
+            checksum: "MAPBOX_PACKAGE_CHECKSUM"),
+        .binaryTarget(
+            name: "MetalANGLE",
+            url: "METAL_ANGLE_PACKAGE_URL",
+            checksum: "METAL_ANGLE_PACKAGE_CHECKSUM")
```
Comment (roblabs) [inline platform/ios/platform/ios/scripts/swift_package_template.swift:24]: For future reviewers (assume Petr already knows this)

Please see notes at https://github.com/maplibre/maplibre-gl-native-distribution/pull/5 on how to test locally built `*.xcframeworks` with SPM.

File: platform/ios/platform/ios/scripts/xcpackage.sh [hunk 3]
```diff
@@ -127,6 +128,16 @@ echo "Creating ${NAME}.xcframework with args: $BUILD_ARGS"
...
+# MetalANGLE
+METAL_ANGLE_NAME="MetalANGLE"
+BUILD_ARGS=""
+BUILD_ARGS="$BUILD_ARGS -output ${BINOUT}/${METAL_ANGLE_NAME}.xcframework"
+addFramework "${BINOUT}/${NAME}-iphoneos.xcarchive" ${METAL_ANGLE_NAME} ${INCLUDE_DEBUG_SYMBOLS}
+addFramework "${BINOUT}/${NAME}-iphonesimulator.xcarchive" ${METAL_ANGLE_NAME} ${INCLUDE_DEBUG_SYMBOLS}
+echo "Creating ${METAL_ANGLE_NAME}.xcframework with args: $BUILD_ARGS"
+xcodebuild -create-xcframework $BUILD_ARGS
+echo "${METAL_ANGLE_NAME}.xcframework created"   
+
```
Comment (roblabs) [inline platform/ios/platform/ios/scripts/xcpackage.sh:140]: This works great!

**Review by roblabs — APPROVED** [review]

**Pull request merged**